src/odemis/driver/xt_client.py

```diff
@@ -1990,3 +2002,178 @@ def stop(self, axes=None):
+        super(MultiBeamScanner, self).__init__(name, role, parent, hfw_nomag, **kwargs)
+
+    def _updateSettings(self):
+        """
+        Read all the current settings from the SEM and reflects them on the VAs
+        """
+        # TODO When the new approach of adding update function to a list is implemented in xt_client.py instead of
+        #  overwriting the method updateSettings the method _updateMBSettings can be added to the list of
+        #  functions to be updated. That means the overwritten part in this class is no longer needed.
+
+        # Polling XT client settings
+        super(MultiBeamScanner, self)._updateSettings()
+        # Polling XTtoolkit settings
+        try:
+            pitch = self.parent.get_pitch()
+            if pitch != self.pitch.value:
+                self.pitch._value = pitch
+                self.pitch.notify(pitch)
+            beam_stigmator = self.parent.get_stigmator()
+            if beam_stigmator != self.beamStigmator.value:
+                self.beamStigmator._value = beam_stigmator
+                self.beamStigmator.notify(beam_stigmator)
+            pattern_stigmator = self.parent.get_pattern_stigmator()
+            if pattern_stigmator != self.patternStigmator.value:
+                self.patternStigmator._value = pattern_stigmator
+                self.patternStigmator.notify(pattern_stigmator)
+            beam_shift_transformation_matrix = self.parent.get_dc_coils()
+            if beam_shift_transformation_matrix != self.beamShiftTransformationMatrix.value:
+                self.beamShiftTransformationMatrix._value = beam_shift_transformation_matrix
+                self.beamShiftTransformationMatrix.notify(beam_shift_transformation_matrix)
+            mpp_rotation = math.radians(self.parent.get_mpp_orientation())
+            if mpp_rotation != self.multiprobeRotation.value:
+                self.multiprobeRotation._value = mpp_rotation
+                self.multiprobeRotation.notify(mpp_rotation)
+            aperture_index = self.parent.get_aperture_index()
+            if aperture_index != self.apertureIndex.value:
+                self.apertureIndex._value = aperture_index
+                self.apertureIndex.notify(aperture_index)
+            beamlet_index = tuple(int(i) for i in self.parent.get_beamlet_index())
+            if beamlet_index != self.beamletIndex.value:
+                self.beamletIndex._value = beamlet_index
+                self.beamletIndex.notify(beamlet_index)
+            multibeam_mode = (self.parent.get_use_case() == 'MultiBeamTile')
+            if multibeam_mode != self.multiBeamMode.value:
+                self.multiBeamMode._value = multibeam_mode
+                self.multiBeamMode.notify(multibeam_mode)
+        except Exception:
+            logging.exception("Unexpected failure when polling XTtoolkit settings")
+
+    def _setPitch(self, pitch):
+        self.parent.set_pitch(pitch * 1e6)  # Convert from micrometer to meters.
```

Comment:
Comment is wrong way around, you're converting from meters to micrometers when calling set_pitch